src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -290,14 +290,27 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                 var clrRundown = new ClrRundownTraceEventParser(source);
                 clrRundown.RuntimeStart += doAtRuntimeStart;
                 source.Clr.RuntimeStart += doAtRuntimeStart;
+                Dictionary<TraceProcess, TraceLoadedDotNetRuntime> processRuntimes = null;
 
                 CircularBuffer<ThreadWorkSpan> RecentThreadSwitches = new CircularBuffer<ThreadWorkSpan>(1000);
                 source.Kernel.ThreadCSwitch += delegate (CSwitchTraceData data)
                 {
```

Comment:
I tried to limit the scope of my implementation since I'm not very familiar with the code. The only reason I found it was through the use of PerfView to analyze opening the GCStats window for a large trace.

The use of `ProcessIndex` sounds promising, but based on @vancem's comments it sounds like it might not work without other changes. I plan to first make the limited corrections that we know are necessary to make this local cache accurate in its current form, and then hopefully we can file an issue to further improve it by using direct lookup instead of a dictionary.